src/rust/engine/ui/src/display.rs

```diff
@@ -32,14 +32,15 @@ use std::collections::{BTreeMap, VecDeque};
 use std::io::Write;
 use std::io::{stdout, Result, Stdout};
 
-use termion::cursor::DetectCursorPos;
 use termion::raw::IntoRawMode;
 use termion::raw::RawTerminal;
+use termion::screen::AlternateScreen;
 use termion::{clear, color, cursor};
 use unicode_segmentation::UnicodeSegmentation;
 
 enum Console {
-  Terminal(RawTerminal<Stdout>),
+  Uninitialized,
```

Comment:
Nit: This is representing console state, which hints that it's possible to change it during runtime. However, the other two values of the enum are not supposed to change once they're set.

Could we represent this as the `EngineDisplay.terminal` field being an `Option<Console>`, so that `None` can represent uninitialised consoles?

This is a small nit, fine to leave it as it is.

Reply:
I have a commit in progress now where we do actually de-initialize and re-initialize the console to fix a bug involving console output during interactive local processes, so I think this representation makes sense.